Triggers.md

```diff
@@ -0,0 +1,234 @@
+   In order to persist the notifications it is possible to write them into a sorted set. 
+   Triggers can provide this ability in a very straightforward way. For example, consider the following function library:
+
+   ![image](https://user-images.githubusercontent.com/88133677/225616988-b9bb8730-59c1-478b-ae21-ece3b8b3a617.png)
+
+   In this implementation we will capture each keyspace event and place the relevant key name and event on a dedicated stream called
+   "keyevent-notifications".
+
+   ```
+   > set a b
+   OK
+   > lpush mylist f
+   (integer) 1
+   > xread streams keyevent-notifications 0
+   1) 1) "keyevent-notifications"
+      2) 1) 1) "1678974201461-0"
+            2) 1) "a"
+               2) "new"
+         2) 1) "1678974201461-1"
+            2) 1) "a"
+               2) "set"
+         3) 1) "1678974210181-0"
+            2) 1) "mylist"
+               2) "new"
+         4) 1) "1678974210181-1"
+            2) 1) "mylist"
+               2) "lpush"
+   ```   
+   
+   [Valkey keyspace Notifications](https://valkey.io/topics/notifications/) has another major disadvantage: they are not reported on the cluster-bus. While this might have good reasoning (eg avoid cluster-bud load, duplication of events etc...), this makes Valkey clients almost unable to manage key events listeners on Valkey clusters. Valkey triggers can enable the user to decide to publish keyspace events on the cluster bus. This way the trigger code can make smart decisions about publishing of the event and then user the 'PUBLISH' call in order to make the event propagated to all cluster nodes.
+
+3. ***Security.*** 
+   In some cases there is a need to limit the access to the cache. Currently Valkey provides restricting functionality via [Access Control Lists](https://valkey.io/topics/acl/).
+   However what if there is a need to access restricted parts of the dataset without granting real permissions to clients?
+   Since a trigger can operate without explicit user request, it can potentially be operated under any defined ACL user which will enable it to operate actions on restricted parts of the dataset.   
+
+### Can't the same be done with modules?
+Modules have the ability to build logic placed on top of keyspace events. However Modules provide a much more restrictive form of logic which is much harder to implement and dynamically replace. 
+By building this new logic inside the redis function library users will have a much more simple way to create stored procedures which are persistent, managed and replicated. 
+
+
+## Design and Specification
+
+### Triggers as part of function libraries
+Valkey triggers will be implemented as an extension of the Valkey [Function Libraries](https://valkey.io/topics/functions-intro/) .
+A trigger is basically a Valkey function which will be called when some specific keyspace event was issued.
```

Comment:
SO I thought of the keyspace events since they are already defined and known to users, so I think it would make it easier for users to adopt. I am open to other suggestions.